src/Dapr.Workflow/Workflow.cs

```diff
@@ -0,0 +1,137 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2023 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+namespace Dapr.Workflow
+{
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Common interface for workflow implementations.
+    /// </summary>
+    /// <remarks>
+    /// Users should not implement workflows using this interface, directly.
+    /// Instead, <see cref="Workflow{TInput, TOutput}"/> should be used to implement workflows.
+    /// </remarks>
+    public interface IWorkflow
```

Comment:
If we don't want users to use this, why not make it internal or just provide the abstract class?

Reply:
I can't make it internal because there are public APIs that depend on it (like the `RegisterWorkflow<T>()` API). I believe there was some reason I needed this because of constraints related to how generics work in .NET, though I can't remember exactly what it was...